### PR TITLE
replace vendored ledger code with npm package

### PR DESCRIPTION
**Pull request by bucko13**
Also includes a more robust check to see if legacy app is being used for ledger version checking.

I confirmed that this works in Caravan with the two signing tests we know work now (p2sh and p2wsh. nested segwit still 
doesn't work with ledger's new app)

Screenshot 2023-03-22 at 12 06 59 PM
Screenshot 2023-03-22 at 12 08 16 PM

### CODE REVIEW

File: src/ledger.ts [hunk 1]
```diff
@@ -232,13 +234,17 @@ export class LedgerInteraction extends DirectKeystoreInteraction {
...
       this.appVersion = response.version;
+      this.appName = response.name;
       return this.appVersion;
     });
   }
 
   async isLegacyApp(): Promise<boolean> {
     const version = await this.setAppVersion();
     const [majorVersion, minorVersion] = version.split(".");
+    // if the name includes "Legacy" than it is legacy app
```
Comment (jevidon) [inline src/ledger.ts:245]: "than" should be "then".

File: src/ledger.ts [hunk 2]
```diff
@@ -232,13 +234,17 @@ export class LedgerInteraction extends DirectKeystoreInteraction {
...
+      this.appName = response.name;
       return this.appVersion;
     });
   }
 
   async isLegacyApp(): Promise<boolean> {
     const version = await this.setAppVersion();
     const [majorVersion, minorVersion] = version.split(".");
+    // if the name includes "Legacy" then it is legacy app
+    if (this.appName && Boolean(this.appName.includes("Legacy"))) return false;
```
Comment (tylerchambers) [inline src/ledger.ts:246]: should this return true?

**Review by tylerchambers — CHANGES_REQUESTED** [review]

File: src/ledger.ts (hunk 2 above)
Comment (bucko13) [reply src/ledger.ts:246]: Ooof. Yes. Not enough coffee for me and not enough storage in my ledger to have both versions of the apps installed apparently

Comment (bucko13) [reply src/ledger.ts]: Fixed. derp. Thanks for catching

**Review by tylerchambers — APPROVED** [review]
I think this is fine now

**Comment by bucko13** [comment]
@tylerchambers I caught up with master and it dismissed your review. Mind giving it another peek?

**Review by tylerchambers — APPROVED** [review]
still looks good!

**Pull request merged**